### ISSUE TITLE
[javascript] cannot parse default values in method parameters

### ISSUE THREAD

**Issue by uhafner** [pmd/pmd#6585]
**Affects PMD Version:**  7.23.0

**Description:**

Parsing a JS method that contains a default value produces a warning. 

Input:
```js
function printPercentage(value, minimumFractionDigits = 2) {
    const options = {
        style: 'percent',
        minimumFractionDigits: minimumFractionDigits
    };
    return Number(value / 100.0).toLocaleString(undefined, options);
}
````

**Log:**

```
[INFO] PMD version: 7.23.0
[WARNING] 1 javascript problems found:
/Users/hafner/git/warnings-ng-plugin-devenv/codingstyle/src/main/js/pmd.js:1:55: missing ) after formal parameters
```

When I use this code in a bigger JS file, additional unrelated warnings show up, somehow the parser gets out of sync:

```
[WARNING] 5 javascript problems found:
/Users/hafner/git/warnings-ng-plugin-devenv/coverage-plugin/plugin/src/main/resources/io/jenkins/plugins/coverage/metrics/steps/CoverageViewModel/view-model.js:45:59: missing ) after formal parameters
/Users/hafner/git/warnings-ng-plugin-devenv/coverage-plugin/plugin/src/main/resources/io/jenkins/plugins/coverage/metrics/steps/CoverageViewModel/view-model.js:45:59: missing } after function body
/Users/hafner/git/warnings-ng-plugin-devenv/coverage-plugin/plugin/src/main/resources/io/jenkins/plugins/coverage/metrics/steps/CoverageViewModel/view-model.js:50:9: invalid return
/Users/hafner/git/warnings-ng-plugin-devenv/coverage-plugin/plugin/src/main/resources/io/jenkins/plugins/coverage/metrics/steps/CoverageViewModel/view-model.js:51:5: syntax error
/Users/hafner/git/warnings-ng-plugin-devenv/coverage-plugin/plugin/src/main/resources/io/jenkins/plugins/coverage/metrics/steps/CoverageViewModel/view-model.js:650:1: syntax error
``` 

**Steps to reproduce:**

Copy the snippet in an empty file and run PMD.

**Running PMD through:** *Maven*

**Comment by uhafner** [comment]
See https://github.com/jenkinsci/coverage-plugin/pull/724/changes/e71daceb3066ce30c1ef00c8cb790c22a86e6d2a

**Comment by UncleOwen** [comment]
I get the same warning when I go all the way back to PMD 7.0.0. Can you please be more precise where the regression is?

**Comment by uhafner** [comment]
This was a guess as my builds that contained the PMD upgrade fail now. But this might be unrelated, so I removed the regression marker.

**Comment by zbynek** [comment]
This might be fixed by updating Rhino  (https://github.com/pmd/pmd/issues/5509)
